pkg/client/apiutil/restmapper.go

```diff
@@ -182,23 +182,26 @@ func (m *mapper) addKnownGroupAndReload(groupName string, versions ...string) er
 		Group:              metav1.APIGroup{Name: groupName},
 		VersionedResources: make(map[string][]metav1.APIResource),
 	}
-	if _, ok := m.knownGroups[groupName]; ok {
-		groupResources = m.knownGroups[groupName]
-	}
 
 	// Update information for group resources about versioned resources.
 	// The number of API calls is equal to the number of versions: /apis/<group>/<version>.
```

Comment:
Maybe let's add a note here as well that this func cleans up / resets m.apiGroups & m.knownGroups if we encounter a missing apiVersion

And that this will lead to:
* apiGroups => we'll load the APIGroup again on the next call of findAPIGroupByName
* knownGroups => we'll only add the now existing apiVersions further down (and the removed ones are dropped)